File: examples/ts-app/src/processor.ts
export type Report = {
	projectName: string;
	lineCount: number;
	filteredLines: string[];
	apiKeyConfigured: boolean;
};

export const buildReport = (
	projectName: string,
	lines: string[],
	minLineLength: number,
	apiKeyConfigured: boolean,
): Report => {
	const filteredLines = lines.filter((line) => line.length >= minLineLength);
	return {
		projectName,
		lineCount: lines.length,
		filteredLines,
		apiKeyConfigured,
	};
};

File: examples/java-app/src/main/java/example/javaapp/Processor.java
package example.javaapp;

import java.util.List;
import java.util.stream.Collectors;

public final class Processor {

	private Processor() {}

	public record Report(String projectName, int lineCount, List<String> filteredLines) {}

	public static Report buildReport(String projectName, List<String> lines, int minLineLength) {
		List<String> filteredLines = lines.stream()
				.filter(line -> line.length() >= minLineLength)
				.collect(Collectors.toList());
		return new Report(projectName, lines.size(), filteredLines);
	}
}
